src/components/InventoryManager/ListingSteps/UnitsStep.js

```diff
@@ -0,0 +1,396 @@
+              </Typography>
+              &nbsp;
+              <Typography display="inline" className={classes.text}>
+                PER PALLET
+              </Typography>
+            </Grid>
+          </Grid>
+        </Grid>
+        <Grid item container xs={12}>
+          <Typography className={classes.heading}>{questions[3]}</Typography>
+        </Grid>
+        <Grid item container xs={12}>
+          {/* TODO : reset master pallet count when checbox unchecked */}
+          <Typography className={classes.text}>
+            <GreenCheckbox
+              checked={listingRecord['has master pallets']}
+              onChange={() => onSelect('has master pallets', !listingRecord['has master pallets'])}
+            />
+            Yes, I do have master pallets.
+          </Typography>
+        </Grid>
+        {(listingRecord['has master pallets'])
+        && (
+        <Grid container item xs={12}>
+          <Grid container spacing={1} item xs={12}>
+            <Grid item xs={12}>
+              <Typography className={classes.subheading}>
+                Quantity
+              </Typography>
+            </Grid>
+            <Grid container item xs={12} alignItems="center">
+              <ListingInputField
+                id="standard-number"
+                name="pallets per master pallet"
+                type="number"
+                onChange={onChangeField}
+                val={listingRecord['pallets per master pallet']}
+                onButtonClick={onButtonClick}
+              />
+              <Typography display="inline" className={classes.labels}>
+                PALLETS
+              </Typography>
+              &nbsp;
+              <Typography display="inline" className={classes.text}>
+                PER
+              </Typography>
+              &nbsp;
+              <Typography display="inline" className={classes.labels}>
+                MASTER PALLET
+              </Typography>
+            </Grid>
+          </Grid>
+        </Grid>
+        )}
+        <Grid item container xs={12}>
+          <Typography className={classes.heading}>{questions[4]}</Typography>
+        </Grid>
+        <Grid container item xs={12}>
+          <Grid container spacing={1} item xs={12}>
+            <Grid item xs={12}>
+              <Typography className={classes.subheading}>
+                Quantity
+              </Typography>
+            </Grid>
+            <Grid container item xs={12} alignItems="center">
+              <ListingInputField
+                id="standard-number"
+                name="pallets available"
+                type="number"
+                onChange={onChangeField}
+                val={listingRecord['pallets available']}
+                onButtonClick={onButtonClick}
+              />
+              <Typography display="inline" className={classes.labels}>
+                PALLETS
+              </Typography>
```

Comment:
If the user checks "yes i do have master pallets" for number 5 can you add a text input that solicits the number of master pallets they have?